src/lib/components/Swap/index.tsx

```diff
@@ -47,20 +47,25 @@ export interface SwapProps {
 }
 
 export default function Swap(props: SwapProps) {
-  useTokenList(props.tokenList)
+  const list = useTokenList(props.tokenList)
   useSyncSwapDefaults(props)
   useSyncConvenienceFee(props)
 
-  const { active, account } = useActiveWeb3React()
+  const { active, account, chainId } = useActiveWeb3React()
   const [boundary, setBoundary] = useState<HTMLDivElement | null>(null)
 
   const [displayTxHash, setDisplayTxHash] = useAtom(displayTxHashAtom)
   const pendingTxs = usePendingTransactions()
   const displayTx = getSwapTx(pendingTxs, displayTxHash)
 
+  const onSupportedChain = useMemo(
+    () => list.reduce((acc, cur) => acc || cur.chainId === chainId, false),
```

Comment:
This is not necessarily true, because a token list loads asynchronously. It'll need three states - loading, supported, unsupported.